kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs

```diff
@@ -608,12 +608,16 @@ impl<'tcx> GotocCtx<'tcx> {
 
     /// fetch the niche value (as both left and right value)
     pub fn codegen_get_niche(&self, v: Expr, offset: Size, niche_ty: Type) -> Expr {
-        v // t: T
-            .address_of() // &t: T*
-            .cast_to(Type::unsigned_int(8).to_pointer()) // (u8 *)&t: u8 *
-            .plus(Expr::int_constant(offset.bytes_usize(), Type::size_t())) // ((u8 *)&t) + offset: u8 *
-            .cast_to(niche_ty.to_pointer()) // (N *)(((u8 *)&t) + offset): N *
-            .dereference() // *(N *)(((u8 *)&t) + offset): N
+        if offset.bytes() == 0 {
```

Comment:
```suggestion
        if offset == Size::ZERO {
```

Reply:
(I just think it's better to use the Size type than to keep converting from bits to bytes and back.)